app/modbus_client.py

```diff
@@ -87,11 +87,12 @@ def _write_register(self, name: str, value):
             except ModbusException as ex:
                 raise ModbusClientError(ex)
 
-    def write_command(self, name: str, value):
+    def write_command(self, message):
         sent = 0
 
-        coil_configuration = self.configuration.get_coil(name)
-        if coil_configuration:
+        name = message.name
+        value = message.value
+        if message.input_type == "COIL":
```

Comment:
could we use an enum rather than strings? https://docs.python.org/3/library/enum.html
I am not an expert in Python as you know. So my natural go-to is going to what I am familiar.
In general though, I prefer strong types when possible. Doing string matching is something I am not that comfortable with. An Enum makes more comfortable for that reason.

Reply:
yes agree, an enum is much better, will add that